### ISSUE TITLE
Switching between profiles doesn't update the archive list

### ISSUE THREAD

**Issue by herrwusel** [borgbase/vorta#1298]
**Describe the bug**
This happened after updating from 0.8.3 to 0.8.4. When I switch between two profiles (each is a different repo but every other setting is the same), the archive list does not update accordingly. I can't pick an archive from my second profile. The name on top of the list doesn't change as well. It seems like after updating my two profiles are set to one repository instead of each one having their own repo.

**To Reproduce**
1. Start vorta
2. Go to the archive tab
3. Switch profile

**Environment (please complete the following information):**
- OS: Arch Linux
- Vorta version: 0.8.4
- Installed from: AUR

Vorta and Borg versions can be found in Main Window > Misc Tab.

**Additional context**
Edit: here are the log entries when switching profiles.
```
2022-05-17 15:13:37,674 - vorta.views.source_tab - DEBUG - Added item number 0 from 1
2022-05-17 15:13:37,675 - vorta.views.source_tab - DEBUG - Added item number 1 from 2
2022-05-17 15:13:39,083 - vorta.views.source_tab - DEBUG - Added item number 0 from 1
2022-05-17 15:13:39,084 - vorta.views.source_tab - DEBUG - Added item number 1 from 2
2022-05-17 15:13:40,822 - vorta.views.source_tab - DEBUG - Added item number 0 from 1
2022-05-17 15:13:40,823 - vorta.views.source_tab - DEBUG - Added item number 1 from 2
```

**Comment by m3nu** [comment]
We had big UI changes in this version, so I wouldn't be surprised. Still can't reproduce this here. Anyone else seeing this? Anything in the logs (missing in the issue)?

**Comment by herrwusel** [comment]
I fixed it by exporting one of the profiles and somehow the repo was changed. So I changed it back to the previous one and now I can see the archive lists for each repo again.

**Comment by herrwusel** [comment]
Yes, when I exported the profiles, they had the same repo. I had to unlink both repos and add them again to make both profiles work.

**Comment by ubuntusuperuser** [comment]
I'm experiencing a similar bug, however it's when switching repository's instead of profiles.
when I switch repository's (under the "Repository" tab) the archive list doesn't update and is stuck on the last added repository, and when I click "Start Backup" it only backups to the last added repo.

I can reproduce on two systems.
first system:
OS: unraid
using docker image: ghcr.io/borgbase/vorta-docker
Version: 0.8.4
 
second system:
OS: ubuntu 20.04.4 (LTS)
Version 0.8.4
Installed from: Flatpak

**Comment by m3nu** [comment]
Ah. Now I see. That's leftover code, since we used to have actions as part of the menu. Simple fix in #1308.

Thanks everyone for testing! The final update by @ubuntusuperuser pointed me in the right direction.
